packages/plugin-dlx/sources/commands/dlx.ts

```diff
@@ -81,6 +81,14 @@ export default class DlxCommand extends BaseCommand {
         await Configuration.updateConfiguration(tmpDir, current => {
           const nextConfiguration = miscUtils.toMerged(current, dlxConfiguration);
 
+          // if `SettingsType.ABSOLUTE_PATH`'s value is a relative path, we need the path relative to the source configuration.
+          // for convenience, all types of fields use the values parsed by source configuration
+          for (const key of Object.keys(current)) {
+            const value = sourceConfiguration.settings.get(key);
+            if (!value) continue;
+            nextConfiguration[key] = sourceConfiguration.values.get(key);
+          }
```

Comment:
Won't that essentially dump all the configuration, including those taken from the environment, those set by default, and those inherited from the home folder?

Reply:
we copy the project directory's `.yarnrc.yml` to the temporary directory, it will also load all the plugins of the project directory,

so I guess what we want is all the configration based on the project directory 🤔, and I expect it should also be able to get the project directory's `.env` https://github.com/yarnpkg/berry/pull/4835 in the future,

in my opinion this is an easy and comprehensive approach, but I may be wrong 😥

Reply:
Or should I look up all the `SettingsType.ABSOLUTE_PATH` in `SettingsType.SHAPE`, `SettingsType.MAP` in a recursive way, and just replace them.

this feels the most stable😀